tools/src/main/scala/org/scalanative/bindgen/Bindgen.scala

```diff
@@ -0,0 +1,160 @@
+package org.scalanative.bindgen
+
+import java.io.File
+
+import scala.collection.immutable
+import scala.sys.process.Process
+
+sealed trait Bindgen {
+
+  /**
+   * Set bindgen executable
+   */
+  def bindgenExecutable(executable: File): Bindgen
+
+  /**
+   * Set header file for which bindings will be generated
+   */
+  def header(header: File): Bindgen
+
+  /**
+   * Library to link with, e.g. -luv for libuv
+   */
+  def link(library: String): Bindgen
+
+  /**
+   * Name of Scala object that contains bindings.
+   * Default is set to library name.
+   */
+  def scalaObjectName(scalaObjectName: String): Bindgen
+
+  /**
+   * Package name of generated Scala file
+   */
+  def packageName(packageName: String): Bindgen
+
+  /**
+   * Declarations will be removed if their names
+   * contain given prefix
+   */
+  def excludePrefix(prefix: String): Bindgen
+
+  /**
+   * Additional argument to append to the compiler command line
+   */
+  def extraArg(args: String*): Bindgen
+
+  /**
+   * Additional argument to append to the compiler command line
+   */
+  def extraArgBefore(args: String*): Bindgen
+
+  /**
+   * Run binding generator
+   */
+  def generate(): Bindings
+}
+
+object Bindgen {
+  def apply(): Bindgen = Impl()
+
+  private final case class Impl(
+      executable: File = null,
```

Comment:
How about using `Option[File]` and default to `scala-native-bindgen`

```scala
executable.fold("scala-native-bindgen")(_.getAbsolutePath)
```

Reply:
But then we should warn that default value is set because user might simply forget to set `executable`, so I would not use default